core/unix/signal_linux_arm.c

```diff
@@ -109,8 +130,38 @@ signal_frame_extra_size(bool include_alignment)
     return 0;
 }
 
+static void
+vfp_query_signal_handler(int sig, siginfo_t *siginfo, kernel_ucontext_t *ucxt)
+{
+    uint offset = 160; /* size of IWMMXT frame */
+    char *coproc = (char *)&ucxt->coproc;
+    kernel_vfp_sigframe_t *vfp0 = (kernel_vfp_sigframe_t *)coproc;
+    kernel_vfp_sigframe_t *vfp1 = (kernel_vfp_sigframe_t *)(coproc + offset);
+    bool vfp0_good = (vfp0->magic == VFP_MAGIC &&
+                      vfp0->size == sizeof(kernel_vfp_sigframe_t));
+    bool vfp1_good = (vfp1->magic == VFP_MAGIC &&
+                      vfp1->size == sizeof(kernel_vfp_sigframe_t));
+    if (vfp0_good == vfp1_good) {
+        ASSERT(false && "Cannot identify VFP frame offset");
+        /* It is safer not to continue. */
+        exit_process_syscall(1);
```

Comment:
A fatal error and exit needs a release-build message (ASSERT is a nop in release).  Please add a new message to events.mc and use a SYSLOG here.  Use os_terminate() so we go through standard exit code.